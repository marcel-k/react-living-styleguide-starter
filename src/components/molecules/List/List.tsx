import React from 'react';
import { ReactNode } from 'react';
import styled from 'styled-components/macro';

export interface ListProps {
  /**
   * One or more components that either:
   * - are li element
   * - are ListItem component
   * - inherit from ListItem component
   */
  children?: ReactNode; // TODO: howto enforce ListItem or li element
}

/**
 * An unordered List component
 */
const StyledList = styled.ul`
  margin: 0;
  padding: 0;
  border-bottom: 1px solid #ccc; /* TODO: Theme */

  li {
    list-style: none;
  }
`;

const List = (props: ListProps) => {
  const { children } = props;

  return (
    <StyledList>
      {children}
    </StyledList>
  );
};

export default List;