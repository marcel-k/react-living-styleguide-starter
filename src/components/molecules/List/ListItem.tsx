import React from 'react';
import { ReactNode, KeyboardEvent } from 'react';
import styled from 'styled-components/macro';

export interface ListItemProps {
  /**
   * If this listitem is selectable
   * @default true
   */
  selectable?: boolean;
  /**
   * If this listitem is selected
   * @default false
   */
  selected?: boolean;
  /**
   * Fires when the list-item:
   * - is clicked
   * - touched
   * - TODO: enter is pressed when focused
   */
  onSelect?: () => void;
  /**
   * Child elements will be horizontaly aligned in the middle (flex-direction row).
   */
  children?: ReactNode;
}

/* TODO: theme */
/* TODO: these functions are re-useable up to a certain level */
const cursor = ({ selectable }: ListItemProps) => selectable ? 'pointer' : 'inherit';
const color = ({ selected }: ListItemProps) => selected ? '#fff' : 'inherit';
const backgroundColor = ({ selected }: ListItemProps) => selected ? '#0000ff' : 'transparent';
const backgroundColorHover = ({ selectable }: ListItemProps) => selectable ? '#eee' : 'transparent';

/**
 * A simple ListItem component
 */
const StyledListItem = styled.li<ListItemProps>`
  min-height: 40px; /* TODO: theme */
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: left;
  padding: 4px 8px; /* TODO: theme */
  border: 1px solid #ccc; /* TODO: theme */
  border-bottom: 0;
  cursor: ${cursor};
  color: ${color};
  background-color: ${backgroundColor};

  :hover {
    background-color: ${backgroundColorHover};
  }
`;

/**
 * A simple selectable list-item component
 */
const ListItem = (props: ListItemProps) => {
  const { className } = props as any; // TODO: we know this is passed by styled components, but don't want it on our interface?
  const {
    children,
    selected = false,
    selectable = true,
    onSelect = () => { }
  } = props;

  // TODO: refactor to utill function?
  const handleKeyDown = (_event: KeyboardEvent<HTMLLIElement>) => {

    // if (event.key === 'Enter') {
    // TODO: haha, li can't be focused
    //   handleSelect();
    // }
  };

  const handleSelect = () => {
    if(selectable) {
      onSelect();
    }
  }

  return (
    <StyledListItem
      onClick={handleSelect}
      selected={selectable && selected}
      onTouchEnd={handleSelect}
      selectable={selectable}
      onKeyDown={handleKeyDown}
      className={className}
    >
      {children}
    </StyledListItem>
  );
}

export default ListItem;