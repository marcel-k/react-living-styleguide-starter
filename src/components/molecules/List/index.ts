export {default as List } from './List';
export * from './List';
export {default as ListItem } from './ListItem';
export * from './ListItem';
export {default as ListItemCheckable } from './ListItemCheckable';
export * from './ListItemCheckable';