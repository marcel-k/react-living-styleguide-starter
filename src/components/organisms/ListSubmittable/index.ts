export * from './ListSubmittable';
export { default as ListSubmittable } from './ListSubmittable';