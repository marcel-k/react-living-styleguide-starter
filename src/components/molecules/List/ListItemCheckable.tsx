import React, { ChangeEvent } from 'react';
import styled from 'styled-components/macro';

import { CheckBox, CheckBoxProps } from '../../atoms/CheckBox';
import ListItem, { ListItemProps } from './ListItem';

// TODO: is extending interfaces like this a good idea?
export interface ListItemCheckableProps extends ListItemProps, CheckBoxProps {
  /**
   * @default false
   */
  selectable?: boolean;
}

const StyledListItemCheckable = styled(ListItem)`
  padding-left: 0;
`;

/**
 * A simple checkable listitem component
 */
const ListItemCheckable = (props: ListItemCheckableProps) => {
  const {
    children,
    selectable = false,
    onChange = () => { },
    ...passThroughProps
  } = props;

  const handleChange = (event: ChangeEvent<HTMLInputElement>, checked: boolean) => {
    onChange(event, checked);
  };

  return (
    <StyledListItemCheckable selectable={selectable} {...passThroughProps}>
      <div onClick={(e) => { e.stopPropagation(); e.nativeEvent.stopImmediatePropagation()}}>
        <CheckBox {...passThroughProps} onChange={handleChange} />
      </div>
      <div>
        {children}
      </div>
    </StyledListItemCheckable>
  );
}

export default ListItemCheckable;