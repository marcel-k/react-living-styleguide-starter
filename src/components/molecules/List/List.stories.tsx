import { text, boolean } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React, { Dispatch, SetStateAction } from 'react';
import { useState } from 'react';

import List from './List';
import ListItem from './ListItem';
import ListItemCheckable from './ListItemCheckable';

/**
 * @returns array of state, dispatch function and toggle function
 */
const useCheckableIds = (initialState: number[] = []): [number[], Dispatch<SetStateAction<number[]>>, (id: number) => void] => {
  const toggleCheckedId = (id: number) => {
    const index = checkedIds.findIndex((checkedId: number) => checkedId === id);
    if (index !== -1) {
      setCheckedIds(checkedIds.filter((_id, i) => i !== index));
    } else {
      setCheckedIds([...checkedIds, id]);
    }
  };

  const [checkedIds, setCheckedIds] = useState(initialState);

  return [checkedIds, setCheckedIds, toggleCheckedId];
}

storiesOf('molecules/List', module)
  .add('default', () => {
    const [selectedId, setSelectedId] = useState(0);
    return (
      <List>
        {[0, 1, 2].map((id) =>
          <ListItem
            key={id}
            selected={selectedId === id}
            onSelect={() => setSelectedId(id)}
            selectable={boolean('selectable', true)}
          >
            {text(`text ${id}`, `list-item ${id}`)}
          </ListItem>
        )}
      </List>
    );
  })
  .add('single checkable', () => {
    const [checkedId, setCheckedId] = useState(0);

    return (
      <List>
        {[0, 1, 2].map((id) =>
          <ListItemCheckable
            key={id}
            checked={checkedId === id}
            onChange={() => setCheckedId(id)}
          >
            {text(`text ${id}`, `list-item ${id}`)}
          </ListItemCheckable>
        )}
      </List>
    );
  })
  .add('multiple checkable', () => {
    const [checkedIds, _, toggleCheckedId] = useCheckableIds([0]);

    return (
      <List>
        {[0, 1, 2].map((id) =>
          <ListItemCheckable
            key={id}
            checked={checkedIds.some((checkedId) => checkedId === id)}
            onChange={() => toggleCheckedId(id)}
          >
            {text(`text ${id}`, `list-item ${id}`)}
          </ListItemCheckable>
        )}
      </List>
    );
  })
  .add('checkable and selectable', () => {
    const [checkedIds, _, toggleCheckedId] = useCheckableIds([0]);
    const [selectedId, setSelectedId] = useState(1);

    return (
      <List>
        {[0, 1, 2].map((id) =>
          <ListItemCheckable
            key={id}
            selectable={true}
            selected={selectedId === id}
            onSelect={() => setSelectedId(id)}
            onChange={() => toggleCheckedId(id)}
            checked={checkedIds.some((checkedId) => checkedId === id)}
          >
            {text(`text ${id}`, `list-item ${id}`)}
          </ListItemCheckable>
        )}
      </List>
    );
  });