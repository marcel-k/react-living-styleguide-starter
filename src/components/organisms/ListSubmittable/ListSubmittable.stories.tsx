import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { useEffect, useState } from 'react';
import { Dispatch, SetStateAction, ChangeEvent } from 'react';

import { ListItemProps, ListItemCheckableProps } from '../../molecules/List';
import ListItemCheckable from '../../molecules/List/ListItemCheckable';
import ListSubmittable from './ListSubmittable';
import { text } from '@storybook/addon-knobs';

type ListItemPropsWithId = ListItemProps & { id: number };
type ListItemCheckablePropsWithId = ListItemCheckableProps & { id: number };

const serverItems = ['item 0', 'item 1', 'item 2', 'item 3', 'item 4'];
const createListItem = (
  id: number,
  value: string,
  selected: boolean,
  onSelect: Dispatch<SetStateAction<number>>,
  extraProps: {} = {}): ListItemPropsWithId => {
  return {
    ...extraProps,
    id,
    selected,
    children: value,
    onSelect: () => onSelect(id)
  };
};

const createCheckableListItem = (
  id: number,
  value: string,
  checked: boolean,
  onChange: (event: ChangeEvent<HTMLInputElement>, checked: boolean) => void,
  extraProps: {} = {}): ListItemCheckableProps & { id: number } => {
  return {
    ...extraProps,
    id,
    checked,
    onChange,
    children: value
  };
};

const useListItemApi = (initialValue: { id: number, value: string }[] = []) => {
  const [items, setItems] = useState(initialValue);

  useEffect(() => {
    // this would be a fetch from server. ideally, the items from the server would have an id already but the world is not ideal.
    const asyncFetchedItems = serverItems;
    const convertedItems = asyncFetchedItems.map((item, i) => ({ id: i, value: item }));
    setItems(convertedItems);
  }, []);

  // TODO: change to 'as consts' in TypeScript v3.4
  return [items, setItems] as [{ id: number, value: string }[], Dispatch<SetStateAction<{ id: number, value: string }[]>>];
}

const useSelectableListItems = (initialSelectedId = 0, items: { id: number; value: string; }[] = []) => {
  const [selectedId, setSelectedId] = useState(initialSelectedId);
  const [selectableItems, setSelectableItems] = useState<ListItemPropsWithId[]>(items);

  useEffect(() => {
    const convertedItems = items.map((item) => createListItem(item.id, item.value, item.id === selectedId, setSelectedId));
    setSelectableItems(convertedItems);
  }, [items, selectedId]);

  useEffect(() => {
    // only for storybook purposes
    action(`item selected with id ${selectedId}`)();
  }, [selectedId]);

  return { selectedId, setSelectedId, selectableItems, setSelectableItems };
}

const useFilteredListItems = <T extends ListItemPropsWithId>(allItems: T[]) => {
  const [filter, setFilter] = useState('');
  const [filteredItems, setFilteredItems] = useState(allItems);

  const filterItems = (items: T[], query: string = '') => {
    return items.filter(({ children }) => query === '' || (children as string).indexOf(query) !== -1);
  };

  const reset = () => {
    setFilter('');
    setFilteredItems(allItems);
  };

  useEffect(() => {
    setFilteredItems(filterItems(allItems, filter));
  }, [allItems, filter]);

  return { filter, setFilter, filteredItems, setFilteredItems, reset };
}

const useCheckableListItems = (initialCheckedIds = [0], items: ListItemPropsWithId[] = []) => {
  const [checkedIds, setCheckedIds] = useState(initialCheckedIds);
  const [checkableItems, setCheckableItems] = useState<ListItemCheckablePropsWithId[]>(items);

  const toggleChecked = (id: number) => {
    const index = checkedIds.findIndex((checkedId: number) => checkedId === id);
    if (index !== -1) {
      action(`uncheck id ${id}`)();
      setCheckedIds(checkedIds.filter((_id, i) => i !== index));
    } else {
      action(`check id ${id}`)();
      setCheckedIds([...checkedIds, id]);
    }
  };
  const isChecked = (id: number) => checkedIds.some((checkedId) => checkedId === id);
  const onChange = (id: number) => () => toggleChecked(id);

  useEffect(() => {
    const convertedItems = items.map(({ id, children, ...rest }) => createCheckableListItem(id, children as string, isChecked(id), onChange(id), rest));
    setCheckableItems(convertedItems);
  }, [items, checkedIds]);

  return { checkableItems, toggleChecked };
}

storiesOf('organisms/ListSubmittable', module)
  .add('default listitems', () => {
    const [items] = useListItemApi([]);
    const [value, setValue] = useState('');
    const { selectableItems } = useSelectableListItems(0, items);
    const {
      filteredItems,
      setFilter,
      reset
    } = useFilteredListItems(selectableItems);

    const resetAndClear = () => {
      reset();
      setValue('');
    };

    return (
      <ListSubmittable
        value={value}
        listItems={filteredItems}
        buttonLabel={text('buttonLabel', 'Search')}
        label={text('label', 'Searchable item list')}
        onReset={() => { resetAndClear(); action('onReset')() }}
        placeholder={text('placeholder', 'Enter search query..')}
        onSubmit={(val) => { setFilter(val); action('onSubmit')(val); }}
        onChange={(_, val) => { setValue(val); action('onChange')(val); }}
      />
    );
  })
  .add('checkable listitems', () => {
    const [value, setValue] = useState('');
    const [items] = useListItemApi([]);
    const { selectableItems } = useSelectableListItems(0, items);
    const { checkableItems } = useCheckableListItems([], selectableItems);
    const {
      filteredItems,
      setFilter,
      reset
    } = useFilteredListItems(checkableItems);

    const resetAndClear = () => {
      reset();
      setValue('');
    };

    return (
      <ListSubmittable
        value={value}
        listItems={filteredItems}
        listItemComponent={ListItemCheckable}
        buttonLabel={text('buttonLabel', 'Search')}
        label={text('label', 'Searchable item list')}
        onReset={() => { resetAndClear(); action('onReset')() }}
        placeholder={text('placeholder', 'Enter search query..')}
        onSubmit={(val) => { setFilter(val); action('onSubmit')(val); }}
        onChange={(_, val) => { setValue(val); action('onChange')(val); }}
      />
    );
  });
