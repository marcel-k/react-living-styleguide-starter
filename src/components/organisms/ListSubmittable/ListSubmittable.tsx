import React from 'react';
import styled from 'styled-components/macro';

import { Button } from '../../atoms/Button';
import { List, ListItem, ListItemProps } from '../../molecules/List';
import { TextBoxSubmittable, TextBoxSubmittableProps } from '../../molecules/TextBoxSubmittable';

export interface ListSubmittableProps<T extends ListItemProps> extends TextBoxSubmittableProps {
  /**
   * Supply a custom item component that extends ListItemProps.
   * @default ListItem will be used as standard component.
   */
  listItemComponent?: React.FunctionComponent<T>;
  /**
   * The items that should be displayed.
   */
  listItems: (T & { id: number | string })[];
  /**
   * Fires when reset button is clicked.
   */
  onReset?: () => void;
  /**
   * This component cannot have children.
   */
  children?: null;
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;

  > div {
    margin-bottom: 4px;
  }
`;

const SearchBoxWrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: flex-end;

  > button {
    margin-left: 4px;
  }
`;

/**
 * A complex generic list component that can be made searchable / filterable with a TextBoxSubmittable.
 */
const ListSubmittable = <T extends ListItemProps>(props: ListSubmittableProps<T>) => {

  const {
    listItems = [],
    onReset = () => { },
    listItemComponent = ListItem,
    ...textBoxProps } = props;

  const ListItemComponent = listItemComponent;

  return (
    <Wrapper>
      <SearchBoxWrapper>
        <TextBoxSubmittable {...textBoxProps} />
        <Button onClick={onReset} secondary>reset</Button>
      </SearchBoxWrapper>
      <List>
        {
          listItems.map((listItemProps) => (
            <ListItemComponent {...listItemProps} key={listItemProps.id} />
          ))
        }
      </List>
    </Wrapper>
  );
};

export default ListSubmittable;