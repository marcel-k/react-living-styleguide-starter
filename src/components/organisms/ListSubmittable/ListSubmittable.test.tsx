import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import 'jest-styled-components';
import 'react-testing-library/cleanup-after-each';

import ListSubmittable from './ListSubmittable';
import { ListItem } from '../../molecules/List';

// Note!: only test specific use cases for InputSubmittable.
// The rest is being covered by test of the child components.
describe('<ListSubmittable /> spec', () => {
  it('filters the list items', function () {
    const spy = jest.fn();

    const component = (
      <ListSubmittable
        onSubmit={spy}
        value={'item 2'}
        buttonLabel={'Search'}
        label={'Searchable item list'}
        listItems={[{ id: 1, children: 'item 1' }, { id: 2, children: 'item 2' }]}
      />
    );

    const { container, rerender } = render(component);
    const { firstElementChild } = container;
    expect(firstElementChild).toMatchSnapshot();

    const searchButton = (firstElementChild as Element).querySelector('button') as HTMLButtonElement;
    fireEvent.click(searchButton);
    expect(spy).toHaveBeenCalledTimes(1);

    rerender(
      <ListSubmittable
        onSubmit={spy}
        value={'item 2'}
        buttonLabel={'Search'}
        listItemComponent={ListItem}
        label={'Searchable item list'}
        listItems={[{ id: 2, children: 'item 2' }]}
      />
    );
    expect(firstElementChild).toMatchSnapshot();

  });
});