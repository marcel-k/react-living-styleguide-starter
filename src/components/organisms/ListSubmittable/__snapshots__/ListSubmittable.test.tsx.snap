// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListSubmittable /> spec filters the list items 1`] = `
.c6 {
  height: 40px;
  padding: 8px 16px;
  font-size: 0.875rem;
  text-align: center;
  opacity: 1;
  cursor: pointer;
  pointer-events: inherit;
  color: #fff;
  background-color: #0000ff;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-variant: small-caps;
}

.c6:active {
  opacity: 0.5;
}

.c7 {
  height: 40px;
  padding: 8px 16px;
  font-size: 0.875rem;
  text-align: center;
  opacity: 1;
  cursor: pointer;
  pointer-events: inherit;
  color: #bbb;
  background-color: transparent;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-variant: small-caps;
  border: 0;
}

.c7:active {
  opacity: 0.5;
}

.c8 {
  margin: 0;
  padding: 0;
  border-bottom: 1px solid #ccc;
}

.c8 li {
  list-style: none;
}

.c9 {
  min-height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  padding: 4px 8px;
  border: 1px solid #ccc;
  border-bottom: 0;
  cursor: pointer;
  color: inherit;
  background-color: transparent;
}

.c9:hover {
  background-color: #eee;
}

.c4 {
  margin: 0;
  color: #333;
  font-size: 0.875rem;
  font-weight: 700;
}

.c5 {
  box-sizing: border-box;
  height: 40px;
  padding: 0 8px;
  font-size: 14px;
  text-align: left;
  color: #333;
  border: 1px solid #bbb;
  opacity: 1;
  pointer-events: inherit;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 4px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > div {
  margin-bottom: 4px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c1 > button {
  margin-left: 4px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <label
          class="c4"
        >
          Searchable item list
        </label>
        <input
          class="c5"
          placeholder=""
          type="text"
          value="item 2"
        />
      </div>
      <button
        class="c6"
      >
        Search
      </button>
    </div>
    <button
      class="c7"
    >
      reset
    </button>
  </div>
  <ul
    class="c8"
  >
    <li
      class="c9"
    >
      item 1
    </li>
    <li
      class="c9"
    >
      item 2
    </li>
  </ul>
</div>
`;

exports[`<ListSubmittable /> spec filters the list items 2`] = `
<div
  class="ListSubmittable__Wrapper-sc-1cp02ee-0 iLCtLv"
>
  <div
    class="ListSubmittable__SearchBoxWrapper-sc-1cp02ee-1 eKKfMG"
  >
    <div
      class="TextBoxSubmittable__Wrapper-rmdr1o-0 jDnOGN"
    >
      <div
        class="TextBoxSubmittable__TextBoxWrapper-rmdr1o-1 hbGyev"
      >
        <label
          class="Label-sc-17u3nw2-0 eoSOaH"
        >
          Searchable item list
        </label>
        <input
          class="TextBox__TextBoxDecl-ctezws-0 TextBox-ctezws-1 kwTjsu"
          placeholder=""
          type="text"
          value="item 2"
        />
      </div>
      <button
        class="Button-qd808s-0 cRaRAV"
      >
        Search
      </button>
    </div>
    <button
      class="Button-qd808s-0 nQEmX"
    >
      reset
    </button>
  </div>
  <ul
    class="List__StyledList-sc-18xn1fm-0 kSCmid"
  >
    <li
      class="ListItem__StyledListItem-hhotwc-0 dXmNN"
    >
      item 2
    </li>
  </ul>
</div>
`;
